pyproject.toml

```diff
@@ -1,41 +1,51 @@
-[project]
+[tool.poetry]
 name = "ngsPETSc"
-version = "0.0.5"
-description = "NGSolve/Netgen interface to PETSc."
-readme = "README.md"
+version = "0.0.7a7"
+description = "NGSolve/Netgen interface to PETSc"
 authors = [
-  {name = "Umberto Zerbinati", email = "umberto.zerbinati@maths.ox.ac.uk"},
-  {name = "Patrick E. Farrell", email = "patrick.farrell@maths.ox.ac.uk"},
-  {name = "Stefano Zampini", email = "stefano.zampini@kaust.edu.sa"},
-  {name = "Jack Betteridge", email = "J.Betteridge@imperial.ac.uk"},
+  "Umberto Zerbinati <umberto.zerbinati@maths.ox.ac.uk>",
+  "Patrick E. Farrell <patrick.farrell@maths.ox.ac.uk>",
+  "Stefano Zampini <stefano.zampini@kaust.edu.sa>",
+  "Jack Betteridge <J.Betteridge@imperial.ac.uk>",
 ]
 maintainers = [
-  {name = "Umberto Zerbinati", email = "umberto.zerbinati@maths.ox.ac.uk"},
+  "Umberto Zerbinati <umberto.zerbinati@maths.ox.ac.uk>",
 ]
-license = {file = "LICENSE.txt"}
-dependencies = [
-  "mpi4py",
-  "numpy",
-  "scipy",
+readme = "README.md"
+license = "MIT"
+packages = [
+    {include = "ngsPETSc"}
 ]
-requires-python = ">=3.8"
+documentation = "https://ngspetsc.readthedocs.io/en/latest/"
+repository = "https://github.com/NGSolve/ngsPETSc"
 classifiers = [
   "Programming Language :: Python :: 3",
   "License :: OSI Approved :: MIT License",
   "Operating System :: OS Independent",
   "Development Status :: 3 - Alpha",
 ]
 
-[project.urls]
-Documentation = "https://ngspetsc.readthedocs.io/en/latest/"
-Repository = "https://github.com/NGSolve/ngsPETSc"
+[tool.poetry.dependencies]
+python = "^3.9"
+netgen-mesher = "^6.2"
+netgen-occt = "^7.8"
+petsc4py =  "^3.22.1"
+mpi4py = "^4"
+numpy = "^2"
+scipy = "^1"
 
-[project.optional-dependencies]
-dev = [
-    "pytest",
-    "pylint",
-]
+[tool.poetry.group.dev.dependencies]
+pytest = "^8.3"
+pylint = "^3.3"
+
+[tool.poetry.group.ngsolve.dependencies]
```

Comment:
Why this is optional? ngsPETSc is a package that offers PETSc solvers to ngsolve, it shouldn't be optional

Reply:
ngsPETSc at its bare minimum, only provides netgen meshes to PETSc and vice-versa, which only requires netgen for this reason I think ngsolve can be interpreted as an optional dependency while netgen is not optional.
This also allows people who want to use ngsPETSc only to get netgen mesh and PETSc dmplex not to install the full ngsolve which is a much larger package than netgen.

Reply:
This also works around the issue that netgen packages have a completely different name when installed from source and the explanation I got after questioning this code:
https://github.com/NGSolve/ngsPETSc/blob/main/setup.py

Reply:
I thought the business of ngsPETSc is to provide parallel solver support for ngsolve, not meshes, see also https://docu.ngsolve.org/latest/i-tutorials/index.html#mpi-parallelization-and-cuda-support.